### PR TITLE
Bugfix/healthcheck and startup fixes

### PR DESCRIPTION
**Pull request by khatton-ft**
- Allow the app to start up, even if it can't connect to ES. This allows the healthcheck to return useful information.
- Add `/__build-info` endpoint, correct healthcheck panic guide URLs, add license file
- Refactor into packages, only the application startup code is in package main
- Add smoke tests for writing/reading ES and run ES within CircleCI.

### CODE REVIEW

File: health/healthcheck.go [hunk 1]
```diff
@@ -40,20 +41,20 @@ func (service *healthService) healthChecker() (string, error) {
...
 }
 
-func (service *healthService) connectivityChecker() (string, error) {
+func (service *HealthService) connectivityChecker() (string, error) {
 
-	_, err := (*service.esHealthService).getClusterHealth()
+	_, err := service.esHealthService.GetClusterHealth()
 	if err != nil {
 		return "Could not connect to elasticsearch", err
 	}
```
Comment (galiri) [inline health/healthcheck.go]: How does this connectivity healthcheck give different information from the cluster health one?

Comment (khatton-ft) [inline health/healthcheck.go]: This check doesn't inspect the return value, only that the request was successful (hence the underscore for the first return vaue). The cluster health looks at the value returned.

Comment (galiri) [inline health/healthcheck.go]: I saw that however one is implying an error is a connectivity issue and the other that an error is 
 a 'general health' issue. the call  taht generates the error is the same the error could be the same.

File: main.go [hunk 2]
```diff
@@ -86,50 +91,42 @@ func main() {
...
-		}
-		if err != nil {
-			log.Fatalf("Creating elasticsearch client failed with error=[%v]\n", err)
-		}
-
-		bulkProcessor, err := newBulkProcessor(elasticClient, &bulkProcessorConfig)
-		if err != nil {
-			log.Fatalf("Creating bulk processor failed with error=[%v]\n", err)
-		}
+		ecc := make(chan *elastic.Client)
```
Comment (BerniVarga) [inline main.go]: I love this!

File: service/es_service.go [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+	log "github.com/Sirupsen/logrus"
+	"gopkg.in/olivere/elastic.v3"
+)
+
+var (
+	ErrNoElasticClient error = errors.New("No ElasticSearch client available")
+)
+
+type esService struct {
+	sync.RWMutex
```
Comment (BerniVarga) [inline service/es_service.go]: Is this Mutex really necessary? We are setting all the fields below only once - we never actually change them (also, they are thread safe at their end)

Comment (khatton-ft) [inline service/es_service.go]: If I take it out, `go test -race` finds a data race, because the elastic client is set from a separate goroutine.

Comment (BerniVarga) [inline service/es_service.go]: Hm... i see...

**Pull request merged**